lib/stages/bufferBadInputs.cpp

```diff
@@ -0,0 +1,81 @@
+#include "bufferBadInputs.hpp"
+
+#include "configUpdater.hpp"
+#include "visUtil.hpp"
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::configUpdater;
+using kotekan::Stage;
+
+REGISTER_KOTEKAN_STAGE(bufferBadInputs);
+
+bufferBadInputs::bufferBadInputs(Config& config_, const string& unique_name,
+                                   bufferContainer& buffer_container) :
+    Stage(config_, unique_name, buffer_container, std::bind(&bufferBadInputs::main_thread, this)) {
+
+    uint32_t num_elements = config.get<uint32_t>(unique_name, "num_elements");
+    input_mask_len = sizeof(uint8_t) * num_elements;
+
+    auto input_reorder = parse_reorder_default(config, unique_name);
+    input_remap = std::get<0>(input_reorder);
+
+    out_buf = get_buffer("bad_inputs_buffer");
+    register_producer(out_buf, unique_name.c_str());
+    
+    // Listen for bad input list updates
+    string badInputs = config.get<std::string>(unique_name, "updatable_config/bad_inputs");
+    if (badInputs.length() > 0)
+        configUpdater::instance().subscribe(
+            badInputs, std::bind(&bufferBadInputs::update_bad_inputs_callback, this, std::placeholders::_1));
+}
+
+bufferBadInputs::~bufferBadInputs() {}
+
+bool bufferBadInputs::update_bad_inputs_callback(nlohmann::json& json) {
+    
+    static uint32_t out_buffer_ID = 0;
```

Comment:
Don't use `static` in a stage unless there is very good reason, use a member variable.  Otherwise you cannot have more than one of a given stage.